app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/uploadsupportingdocs/UploadSupportingEvidenceController.scala

```diff
@@ -0,0 +1,1012 @@
+          case CompleteUploadSupportingEvidenceAnswers(
+              doYouWantToUploadSupportingDocumentAnswer,
+              supportingDocuments
+              ) =>
+            CompleteUploadSupportingEvidenceAnswers(doYouWantToUploadSupportingDocumentAnswer, supportingDocuments)
+        }
+        val newDraftReturn = fillingOutReturn.draftReturn match {
+          case s @ DraftSingleDisposalReturn(
+                _,
+                _,
+                _,
+                _,
+                _,
+                _,
+                _,
+                _,
+                _,
+                _,
+                _
+              ) =>
+            s.copy(uploadSupportingDocuments = Some(updatedAnswers))
+          case m @ DraftMultipleDisposalsReturn(
+                _,
+                _,
+                _,
+                _,
+                _,
+                _,
+                _
+              ) =>
+            m.copy(uploadSupportingDocuments = Some(updatedAnswers))
+        }
+
+        val result = for {
+
+          _ <- returnsService
+                .storeDraftReturn(
+                  newDraftReturn,
+                  fillingOutReturn.subscribedDetails.cgtReference,
+                  fillingOutReturn.agentReferenceNumber
+                )
+          _ <- EitherT(
+                updateSession(sessionStore, request)(
+                  _.copy(journeyStatus = Some(fillingOutReturn.copy(draftReturn = newDraftReturn)))
+                )
+              )
+        } yield ()
+
+        result.fold({ e =>
+          logger.warn("Could not update session", e)
+          errorHandler.errorResult()
+        }, _ => Redirect(controllers.returns.routes.TaskListController.taskList()))
+      }
+    }
+
+  @SuppressWarnings(Array("org.wartremover.warts.NonUnitStatements", "org.wartremover.warts.Any"))
+  private def checkYourAnswersHandler(
+    answers: UploadSupportingEvidenceAnswers,
+    fillingOutReturn: FillingOutReturn,
+    draftReturn: DraftReturn
+  )(implicit request: RequestWithSessionData[_]): Future[Result] =
+    answers match {
+      case IncompleteUploadSupportingEvidenceAnswers(None, _) =>
+        Redirect(routes.UploadSupportingEvidenceController.doYouWantToUploadSupportingDocuments())
+      case IncompleteUploadSupportingEvidenceAnswers(Some(doYouWantToUploadSupportingEvidence), supportingEvidences) =>
+        if (doYouWantToUploadSupportingEvidence) {
+          supportingEvidences match {
+            case Nil => Redirect(routes.UploadSupportingEvidenceController.uploadSupportingEvidence())
+            case ::(_, _) =>
+              val uploadedSupportingEvidences = for {
+                documents <- upscanService.getAll(DraftReturnId(draftReturn.id.toString))
+              } yield documents
+              uploadedSupportingEvidences.fold(
+                e => InternalServerError(s"could not get all uploaded supporting evidence: $e"),
```

Comment:
we shouldn't be printing this kind of thing out to the browser - log the error and return `errorHandler.errorResult()`

Reply:
yes that needs to be fixed now - will do that

Reply:
fixed